src/core/a-entity.js

```diff
@@ -539,7 +539,7 @@ var proto = Object.create(ANode.prototype, {
    */
   setAttribute: {
     value: function (attr, value, componentPropValue) {
-      var isDebugMode = this.sceneEl && this.sceneEl.getAttribute('debug');
+      var isDebugMode = this.sceneEl && this.sceneEl.hasAttribute('debug');
```

Comment:
I think the proper solution is that for singleProperty components if the attribute is an empty string we should return the default value that in case of the debug component is `true`